src/commands/ml-models-delete.ts

```diff
@@ -1,9 +1,33 @@
+import * as clc from "cli-color";
 import { Command } from "../command";
+import * as getProjectId from "../getProjectId";
+import { requirePermissions } from "../requirePermissions";
+import { ensureFirebaseMlApiEnabled, isValidModelId } from "../ml/mlHelper";
+import * as mlApi from "../ml/mlApi";
+import * as utils from "../utils";
 
 export default new Command("ml:models:delete <modelId>")
-  .description("deletes the Firebase ML Model with the given modelId")
+  .description("delete the Firebase ML Model with the given modelId")
+  .help("TODO")
+  .before(requirePermissions, ["firebaseml.models.delete"])
+  .before(ensureFirebaseMlApiEnabled)
   .action(
-    async (modelId: string): Promise<void> => {
-      throw new Error("Not Implemented");
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async (modelId: string, options: any): Promise<void> => {
+      if (!isValidModelId(modelId)) {
+        return utils.reject("Must specify a valid model ID", { exit: 1 });
+      }
+      const projectId = getProjectId(options);
+      try {
+        await mlApi.deleteModel(projectId, modelId);
+      } catch (err) {
+        if (err.status === 404) {
+          return utils.reject(`No model ${modelId} in project ${projectId}.`, {
+            exit: 1,
+          });
+        }
+        throw err;
```

Comment:
as a note: you don't always have to throw FirebaseErrors, re-throwing this error is acceptable too (and handling a 404 as above is fine)

Reply:
noted.